core/src/main/java/com/graphhopper/routing/util/AbstractWeighting.java

```diff
@@ -32,6 +38,40 @@ public AbstractWeighting( FlagEncoder encoder )
             throw new IllegalStateException("Not a valid name for a Weighting: " + getName());
     }
 
+
+    @Override
+    public double calcWeight( EdgeIteratorState edgeState, boolean reverse, int prevOrNextEdgeId )
+    {
+        double weight = calcPureWeight(edgeState, reverse,prevOrNextEdgeId);
+
+        if(adjustments != null){
```

Comment:
Yes I thought about it and thought that a null check is faster then querying `size`.